Colony Drop (code only)

GunHog opened the pull request:
Adds the backbone for @KorPhaeron's colony drop feature, without the map changes. This will allow @WJohn to create a map change to his standards.

There are still some changes I can probably make to this code though.

To Do:
- [x] Find some way to check that the drop zone is fully within only one area (so as not to breach ruins / necropolis)
- [x] Have a confirmation prompt for the targeter, OR allow the targeter to be used multiple times until the colony itself is launched. **EDIT:** This is new a do_after

Original PR: https://github.com/tgstation/tgstation/pull/19602

imtakingabreakdontatme committed:
Adds disguting tumours to meta and box, lets RD and engineering build in those tumours

imtakingabreakdontatme committed:
Fixes mining shuttle breaking

imtakingabreakdontatme committed:
Can't launch the colony as the clown

Commit:
Merge remote-tracking branch 'remotes/KorPhaeron/colony_drop' into Colony_Drop_Revival

Commit:
Map change reversions

Xhuis commented on code/modules/mining/mine_items.dm:
```diff
@@ -500,3 +500,72 @@
+
+/obj/item/device/assault_pod/mining/attack_self(mob/living/user)
+	var/turf/T = get_turf(user)
+	if(T.z != ZLEVEL_MINING)
```
You should totally add an admin override variable so players could trade their TC to drop a base on the station.

GunHog replied:
If I can work out the best way to handle that (I am thinking of giving it a prompt or a menu for confirmation), I would be more than happy to do that!

GunHog commented on code/modules/mining/mine_items.dm:
```diff
@@ -500,3 +500,72 @@
+	icon = 'icons/obj/device.dmi'
+	desc = "Deploy to designate the landing zone of the auxillary base."
+	shuttle_id = "colony_drop"
+	dwidth = 11
```
I need to know how to allow the mappers to change these dimensions, so they can size their colonies to best fit their own maps.

KorPhaeron replied:
edit vars on the targeter

Commit:
Tweaks

- Adds Xhuis' suggestion for a badmin version
- Restricts dropping colonies on ruins without being fully inside them.

Commit:
Whoops

>Gun Hog in charge of math

GunHog commented:
There is now a badmin version for you, Xhuis. Also, I added a way to check that the colony does not intersect more than one area - so you have to be completely inside a sufficiently large ruin if you want to drop a base there.

Obviously the mapped colonies will not be as ROFLHUGE as the original test merge one was.

Dorsisdwarf commented:
Does this let me set any area to be colony-dropped, so I can badmin colony-drop shitcurity into the gulag?

KorPhaeron commented:
no

Commit:
Adds Do_After and Announcement

- Adds Do_After as comfirmation of setting the colony's location.
- Adds an announcement for when the drop zone is set.

GunHog commented:
New in this commit:
- Adds Do_After as comfirmation of setting the colony's location.
- Adds an announcement for when the drop zone is set.

optimumtact commented:
is this ready?

GunHog commented:
Yes. Any additional adjustments I can do once @WJohn sends me his map update.
